core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java

```diff
@@ -57,7 +57,7 @@
 
     private static final Pattern<FilterNode> PATTERN =
             filter().with(source().matching(
-                    tableScan().capturedAs(TABLE_SCAN)));
+                    tableScan().capturedAs(TABLE_SCAN).matching(node -> !node.getEnforcedConstraint().isAll())));
```

Comment:
The rule code below doesn't make it obvious that this is a special case..
Maybe rename the rule to `RemoveRedundantPredicateAboveTableScan`